packages/xo-server/src/xo-mixins/migrate-vm.mjs

```diff
@@ -288,10 +290,16 @@ export default class MigrateVm {
             const { fileName, path, datastore, isFull } = disk
             const { vdi, vhd: parentVhd } = vhds[userdevice]
             let vhd
+            if (vdi === undefined) {
+              throw new Error(`Can't import delta of a running VM without its parent vdi`)
```

Comment:
To whom is this message intended? XO support/devs or users?

If it's intended for customers, is it clear enough?

Reply:
Do you think it's not clear enough ?  this shouldn't happen, but, if it happens, I would like ta have a better message than `Cannot read properties of undefined`

Reply:
It depends of the answer to my first question :slightly_smiling_face:

Reply:
this is for both, the user need to know that the import failed ( and also trigger all the cleanup on defer) , and the support will be faster with a real error

Reply:
I'm not really sure how it can occur (maybe by adding a disk before making the last snapshot). How would you present it ?

Reply:
After discussing directly, these errors should not happen in normal situations and are intended to the devs/support.

The messages are thus good enough.